pkg/util/deployment/deployment.go

```diff
@@ -293,14 +293,17 @@ func updatePodWithRetries(podClient unversionedcore.PodInterface, pod *api.Pod,
 	var err error
 	oldPod := pod
 	err = wait.Poll(10*time.Millisecond, 1*time.Minute, func() (bool, error) {
+		pod, err = podClient.Get(oldPod.Name)
+		if err != nil {
+			return false, err
+		}
 		// Apply the update, then attempt to push it to the apiserver.
 		applyUpdate(pod)
```

Comment:
applyUpdate should be able to return an error. It should be checking preconditions for the update. We shouldn't blindly retry.